src/test/lib/ApiTest.cpp

```diff
@@ -1152,6 +1152,74 @@ DummyStreamCallback(
     return QUIC_STATUS_SUCCESS;
 }
 
+struct CloseFromCallbackContext {
+    uint16_t CloseCount;
+    volatile uint16_t CurrentCount;
+
+    static QUIC_STATUS StreamCallback(_In_ MsQuicStream*, _In_opt_ void*, _Inout_ QUIC_STREAM_EVENT*) {
+        return QUIC_STATUS_SUCCESS;
+    }
+
+    static QUIC_STATUS Callback(_In_ MsQuicConnection* Conn, _In_opt_ void* Context, _Inout_ QUIC_CONNECTION_EVENT* Event) {
+        auto Ctx = (CloseFromCallbackContext*)Context;
+
+        if (Event->Type == QUIC_CONNECTION_EVENT_PEER_STREAM_STARTED) {
+            new(std::nothrow) MsQuicStream(Event->PEER_STREAM_STARTED.Stream, CleanUpAutoDelete, StreamCallback, nullptr);
+        }
+        // auto count = ++Ctx->CurrentCount;
+#if _WIN32
+        auto count = InterlockedIncrement16((volatile short*)&Ctx->CurrentCount);
+#else
+        auto count = __sync_add_and_fetch(&Ctx->CurrentCount, 1);
```

Comment:
InterlockedIncrement16 is already defined in quic_platform_posix.h for other OSes.